### ISSUE TITLE
Sending credentials to be able to download contents of a private playlist

### ISSUE THREAD

**Issue by MorenoGordo** [pytube/pytube#914]
The "Playlist" class fails when you try to download the videos in a private playlist. 

That's right, obviously. However, I thought it would be a good idea to add a feature to send credentials for YT to allow you to access your own private playlists.

Is this possible?

Thanks.

**Comment by tfdahlin** [comment]
This might be possible to implement, but it will probably be challenging to do. When I have some more free time, I'll look into how we might be able to accomplish this.

It's worth noting, however, that YouTube may crack down on users trying to do this and potentially ban their accounts. By linking your downloads to your youtube account, you would inherently risking the possibility of your account being locked.

**Comment by glubsy** [comment]
Plenty of other scripts do this already. Parsing a Netscape formatted cookie file is simple, then feeding the appropriate tokens via HTTP request should be trivial. Example:

```python
import requests

def parse_cookie_file(cookiefile):
    cookies = {}
    with open(cookiefile, 'r') as fp:
        content = fp.read()
        for line in content.split('\n'):
            if line.startswith('#'):
                continue
            if 'youtube' in line:
                elements = line.split('\t')
                cookies[elements[-2]] = elements[-1]
    return cookies

req = requests.get(url, headers=headers, cookies=cookies)
```

Throttling request rate could mitigate the risk of getting banned by Youtube.

**Comment by tfdahlin** [comment]
@glubsy yes, reading a file is easy, but it's not directly intuitive for the command-line to specify the path to a cookie file. It's also worth noting that simply reusing an existing youtube cookie may flag your account for suspicious activity. If they see the same cookie being used by your Chrome browser and pytube, the browser fingerprints won't match.

**Comment by tfdahlin** [comment]
This functionality has been implemented in v11.0.0 through the use of OAuth. I'll add documentation soon.

**Comment by sean1985zc** [comment]
Can this be down via "Playlist" interface? Could you provide the documention if possible? thanks
